client/src/components/Collections/ListCollectionCreator.vue

```diff
@@ -131,6 +185,18 @@ function _isElementInvalid(element: HistoryItemSummary) {
         return localize("has been deleted or purged");
     }
 
+    if (filterExtensions.value && datatypeToggle.value !== "all" && element.extension) {
+        if (
+            // is the element's extension not a subtype of any of the required extensions?
+            (datatypeToggle.value === "ext" &&
+                !datatypesMapper.value?.isSubTypeOfAny(element.extension, props.extensions!)) ||
+            // else, does the element's extension have the same parent class as any of the required extensions?
```

Comment:
I don't think this is correct, `isSubTypeOfAny` should take this into account already, otherwise datatype matching in the workflow editor or drag and drop extension validation would be broken. Do you have an example where isSubTypeOfAny returns false and isSubClassOfAny is true ?

Reply:
Initially, I had enforced the usage of `isSubTypeOfAny` and hence, the collection builder would only be filtered for extensions explicitly defined (`isSubTypeOfAny === true`).

But then I realized that, for e.g.: if the required extension for the list is `csv`, if you manually create a list from datasets that are not `csv` but are `tabular` instead, that _will_ be a valid input for the csv list **via implicit conversion**.

Therefore, I looked for a way to be more lenient in the list creation as even if the required extension is `csv` and you have no explicit `csv`s, you can still create a list that could be valid via implicit conversion. I felt that the `datatypesMapper` could be the solution for this?

| Here is an example of this: |
| ---- |
| <video src="https://github.com/user-attachments/assets/ff5a4ae3-31f5-47c4-9e88-c65d20641364" /> |
| Here, you can see that for the `Extension` filter, we only show the csvs in the history (`isSubTypeOfAny`), for `Datatype` we show other `tabular` types that you could still create a collection from (`isSubClassOfAny`), and `All` allows for any extension. Then I show you how if you manually create a list from `tabular`s instead, it can still be a valid input... |

> datatype matching in the workflow editor or drag and drop extension validation would be broken

You make a good point though, because once that collection (from extensions other than the required one) is created, I enforce it as input for the field, and that does not seem right because the implicit conversion didn't even happen at that point (I just let the `tabular` list be an input for the field).


If this filtering is problematic, I can do one of these two things (unless there is any other recommendation):
1. Remove the filter buttons, and enforce only `isSubTypeOfAny`; e.g.: only show csvs for csv requirement, and let the user know: "Hey, you might have other datasets available here that you can convert to csv yourself if you wish to add them to the list..."
2. Remove the filter buttons, and extension filtering entirely: allowing users to make a list from whatever, and then letting `FormData` do the validation itself

_Both of these options would mean removing the logic of `isSubClassOfAny` of course._